code/chacha20/Hacl.Impl.Chacha20.fst

```diff
@@ -96,7 +96,6 @@ val chacha20_init:
 let chacha20_init ctx k n ctr =
   let h0 = ST.get() in
   recall_contents chacha20_constants Spec.chacha20_constants;
-  mut_immut_disjoint ctx chacha20_constants (ST.get ());
```

Comment:
The new lemma has a pattern so it isn't needed. Actually I don't know why the old lemma was called in the old code, since it also had a pattern.

Reply:
I think the pattern used to not work because it was on Lib.Buffer.modifies -- you should never use Lib.Buffer.modifies in your patterns because it doesn't have transitivity, reflexivity, etc. and all the other things that make patterns work; so I wrote the manual call first, then fixed the pattern afterwards following help from Nik